Source/FullScreenPokemon.d.ts

```diff
@@ -21,6 +21,21 @@ declare module FullScreenPokemon {
     }
 
     /**
+     * Container for holding the states of non-primitive objects in the game.
+     */
+    export interface IStateHistoryObject {
+        [i: string]: [any];
+    }
+
+    export interface IStateHistorySaveable {
```

Comment:
> IStateHistorySaveable

Weird name. Pretty sure I asked for this to *not* be the name. Please find another, more concise one.